Docs: Update the meta-data reference for extensions (#1163)
grafana/plugin-tools: docusaurus/docs/reference/metadata.md

| ------------------- | ----------------------- | ------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------- |
| `grafanaDependency` | string                  | **Yes**                                                       | Required Grafana version for this plugin. Validated using https://github.com/npm/node-semver.                                 |
| `grafanaVersion`    | string                  | No                                                            | (Deprecated) Required Grafana version for this plugin, e.g. `6.x.x 7.x.x` to denote plugin requires Grafana v6.x.x or v7.x.x. |
| `plugins`           | [object](#plugins)[]    | No                                                            | An array of required plugins on which this plugin depends.                                                                    |

> | `extensions`        | [object](#extensions)[] | Only if you depend on extensions, e.g. on exposed components. | An object defining your plugins dependencies on extensions like exposed components.                                           |

Reviewer: ```suggestion
| `extensions`        | [object](#extensions)[] | Only if you depend on extensions. | An object defining your plugins dependencies on extensions, such as exposed components.                                           |
```

| `type`    | string | **Yes**  | Possible values are: `app`, `datasource`, `panel`.                                                                                                                              |
| `version` | string | **No**   | This property should use a specific, pinned [SemVer](https://semver.org/) version and not a range. If no version is specified, it will resolve to the latest supported version. |

### extensions

> In case your plugin depends on an exposed component using `usePluginComponent()`, you need to register the id of the exposed component under here, otherwise it won't work.

Reviewer: ```suggestion
If your plugin depends on an exposed component using `usePluginComponent()`, you must register the id of the exposed component here, otherwise it won't work.
```

Reviewer: Perhaps these mentioned functions should link to their info? e.g. https://grafana.com/developers/plugin-tools/reference/ui-extensions#useplugincomponents

Author: That's a great idea!

#### Properties

| Property            | Type     | Required                                                 | Description                                                            |
| ------------------- | -------- | -------------------------------------------------------- | ---------------------------------------------------------------------- |

> | `exposedComponents` | string[] | Only if you are using exposed components in your plugin. | A list of the exposed component ids that you are using in your plugin. |

Reviewer: ```suggestion
| `exposedComponents` | string[] | Only if your plugin uses exposed components. | A list of the exposed component ids that your plugin depends on. |
```

## extensions

> Extensions related meta-info.

Reviewer: ```suggestion
Extensions specific metadata.
```

### Properties

| Property            | Type                         | Required                                     | Description                                            |
| ------------------- | ---------------------------- | -------------------------------------------- | ------------------------------------------------------ |
| `addedComponents`   | [object](#addedcomponents)   | Only if your plugin has component extensions | The list of component extensions added by your plugin. |
| `addedLinks`        | [object](#addedlinks)        | Only if your plugin has link extensinos      | The list of link extensions added by your plugin.      |
| `exposedComponents` | [object](#exposedcomponents) | Only if your plugin exposes components       | The list of components exposed by your plugin.         |
| `extensionPoints`   | [object](#extensionpoints)   | Only if your plugin defines extension points | The list of extension points defined by your plugin.   |

### addedComponents

> This list has to contain all component extensions that your plugin registers to other extension points using `.addComponent()`. **Added components that are not listed in here won't work.**

Reviewer: ```suggestion
This list must contain all component extensions that your plugin registers to other extension points using `.addComponent()`. **Components that are not listed here won't work.**
```

#### Properties

| Property      | Type     | Required | Description                                                                                                           |
| ------------- | -------- | -------- | --------------------------------------------------------------------------------------------------------------------- |

> | `targets`     | string[] | **Yes**  | The list of the extension point ids you would like to register your extension to, e.g. `["grafana/user/profile/tab"]` |

Reviewer: ```suggestion
| `targets`     | string[] | **Yes**  | The extension point ids your plugin registers the extension to, e.g. `["grafana/user/profile/tab"]` |
```

| `title`       | string   | **Yes**  | The title of your component extension.                                                                                |
| `description` | string   | No       | An optional description of your component extensions.                                                                 |

### addedLinks

> This list has to contain all link extensions that your plugin registers to other extension points using `.addLink()`. **Added links that are not listed in here won't work.**

Reviewer: ```suggestion
This list must contain all link extensions that your plugin registers to other extension points using `.addLink()`. **Links that are not listed here won't work.**
```

#### Properties

| Property      | Type     | Required | Description                                                                                                               |
| ------------- | -------- | -------- | ------------------------------------------------------------------------------------------------------------------------- |

> | `targets`     | string[] | **Yes**  | The list of the extension point ids you would like to register your extension to, e.g. `["grafana/dashboard/panel/menu"]` |

Reviewer: ```suggestion
| `targets`     | string[] | **Yes**  | The extension point ids your plugin registers the extension to, e.g. `["grafana/dashboard/panel/menu"]` |
```

| `title`       | string   | **Yes**  | The title of your link extension.                                                                                         |
| `description` | string   | No       | An optional description of your link extensions.                                                                          |

### exposedComponents

> This list has to contain all components that your plugin exposes using `.exposeComponent()`. **Exposed components that are not listed in here won't work.**

Reviewer: ```suggestion
This list must contain all components that your plugin exposes using `.exposeComponent()`. **Components that are not listed here won't work.**
```

| ------------- | ------ | -------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `id`          | string | **Yes**  | An id for your exposed component prefixed with your plugin id. It needs to be unique. A best practice is to also add a version suffix to prevent future breaking changes. E.g.: `myorg-extensions-app/exposed-component/v1` |
| `title`       | string | No       | The title of your exposed component.                                                                                                                                                                                        |
| `description` | string | No       | An optional description of your exposed component.                                                                                                                                                                          |

### extensionPoints

> This list has to contain all extension points that your plugin defines using `usePluginLinks()` or `usePluginComponents()`. **Extension points that are not listed in here won't work.**

Reviewer: ```suggestion
This list must contain all extension points that your plugin defines using `usePluginLinks()` or `usePluginComponents()`. **Extension points that are not listed in here won't work.**
```

#### Properties

| Property      | Type   | Required | Description                                                                                                                                                                                                             |
| ------------- | ------ | -------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | `id`          | string | **Yes**  | An id for your extension point prefixed with your plugin id. It needs to be unique. A best practice is to also add a version suffix to prevent future breaking changes. E.g.: `myorg-extensions-app/extension-point/v1` |

Reviewer: ```suggestion
| `id`          | string | **Yes**  | A unique id for your extension point prefixed with the plugin id. It is recommended to add a version suffix to prevent future breaking changes. E.g.: `myorg-extensions-app/extension-point/v1` |
```
